search-api/src/search_api/utils/logging.py

```diff
@@ -29,3 +39,53 @@ def setup_logging(conf):
         print('Configure logging, from conf:{}'.format(conf), file=sys.stdout)
     else:
         print('Unable to configure logging, attempted conf:{}'.format(conf), file=sys.stderr)
+
+
+class Level(BaseEnum):
+    """Enum for the Business state."""
+
+    CRITICAL = auto()
+    DEBUG = auto()
+    ERROR = auto()
+    INFO = auto()
+    WARNING = auto()
+
+
+def get_logging_flag_name():
+    """Get the name of the FFlag to lookup."""
+    try:
+        return current_app.config.get('OPS_LOGGER_LEVEL')
+    except:  # pylint: disable=bare-except; # noqa: E722, B901
+        return 'ops-{name}-log-level'.format(name=__name__)
+
+
+def set_log_level_by_flag():
+    """Set the logging level if the FF has a different level than what is currently set."""
+    try:
+        flag_name = get_logging_flag_name()
+        flag_value = Flags.value(flag_name, user={'key': flag_name})
+        if flag_value and (level_name := logging.getLevelName(logging.getLogger().level)) \
+                and flag_value != level_name:  # pylint: disable=E0601; linter hates the walrus
+            set_logging_level(flag_value)
+    except Exception:  # noqa: B902
+        return
+
+
+def set_logging_level(level: Union(Level, str)):
+    """Set the logging level of the top logger."""
+    _logger = logging.getLogger()
+
+    match level:
```

Comment:
hey thor I think this is 3.10 only so it will fail in the ci for the testing and the lint probably. I think it will also fail in our deployment because the docker file uses 3.9.5? I'm all for this, but can you update the CI file and the dockerfile to python 3.10?

Reply:
I'll update Docker to 3.10, as that is the current target

Reply:
can you update the CI file as well? It will allow your tests to run

Reply:
totally missed that. thanks for catching!!